### ISSUE TITLE
investigate switching to Ammonite for startup scripts

### ISSUE THREAD

**Issue by DanielFallon** [confluentinc/cp-docker-images#327]
Currently the cp-* docker images include python, bash, java and a heavyweight debian base to facilitate the startup of the confluent platform. We could simplify the code base, decrease the footprint of the docker images, and decrease dependencies by switching to ammonite (a scala repl/script runner) for the setup scripts.

The eventual result would be: 
 - lightweight debian base image (jessie-slim) 
 - openjre
 - ammonite
 - confluent platform

I may be spending some time doing this for some images that I maintain, so I thought I'd float the idea of pushing these changes upstream

**Comment by ewencp** [comment]
What are the savings for this? Most of this seems like it doesn't really have to do with ammonite. Using jessie-slim or similar seems like a fine idea, we'd just need to make sure we install any dependencies that would be in a normal "fat" debian image. We're already using the Zulu JDK.

As for ammonite, I just installed python from a fresh jessie-slim and it uses 24.5 MB more disk space, whereas the ammonite script runner was 37 MB. So it seems ammonite would actually use more disk space?

**Comment by DanielFallon** [comment]
Good catch, I actually hadn't looked at individual layer sizes (just overall, and you are correct that most of the size difference was related to using a smaller debian image)

### Ammonite as a better option
I chose ammonite because it has the file and environment operation primitives that are present in bash, the power and simplicity of syntax present in python, while also being hosted within the JVM which allows it to load the java utilities without creating a separate process.

### Additional reasoning
most of the savings that I saw from using ammonite was just a simplification in code base. I felt that the bash -> python -> java linkage that makes up the current command line utilities is far more fragile than a single code base in a single language. One way to decrease the number of links is to simply remove python--particularly since it is used mostly as a wrapper CLI for the utility jar. 

But I'm not sure that using simply bash and java is the right option either, see the presentation below for more compelling arguments
https://docs.google.com/presentation/d/11vZzXCfAA0aOFAuHA0nAvAzALGFGCH-dqHxx6XMgbk8/view

**Comment by DanielFallon** [comment]
Also to clarify, my company and I are choosing to switch to ammonite scripts for our own build and deployment scripts (which exceed the scope of those present in the cp-* images). I was simply offering the possibility of merging some of my work upstream if confluent would find it useful.
